### PR TITLE
Upgrade operator-sdk

### PR DESCRIPTION
**Pull request by andrewazores**
This upgrades operator-sdk to v0.19.4 .

The upgrade path up to 0.19.x is generally straightforward, with the jump to 1.0.0+ requiring a source layout restructure and some other changes to how images are built and packaged, so I will attempt that separately. There is a recommendation to get up to 0.19.0 first before moving to 1.0.0, so that was my goal here.

There were a series of odd go.mod version conflicts, or "module doesn't contain package" compile errors, along the way, but what I have here seems to work. I don't know how important some of the pseudo-version `replace`s were from before, however.

I have tried testing this in CRC and it seems okay, although during `make test` my CRC instance keeps dying and restarting. I encounter this often however, so I don't think it has to do with this PR in particular, and manually testing it seems to turn up normal results.

### CODE REVIEW

**Comment by andrewazores** [comment]
I finally got CRC to survive a test round:

```
Ginkgo ran 4 suites in 11.893690167s
Test Suite Passed
operator-sdk scorecard
[Deprecation Notice] Operator SDK has a new CLI and project layout that is aligned with Kubebuilder.
See `operator-sdk init -h` and the following doc on how to scaffold a new project:
https://v0-19-x.sdk.operatorframework.io/docs/golang/quickstart/
To migrate existing projects to the new layout see:
https://sdk.operatorframework.io/docs/building-operators/golang/project_migration_guide/

INFO[0000] Using config file: /home/andrew/workspace/container-jfr-operator/.osdk-scorecard.yaml 
I0202 14:07:29.099943  168669 request.go:621] Throttling request took 1.079311997s, request: GET:https://api.crc.testing:6443/apis/rhjmc.redhat.com/v1beta1?timeout=32s
I0202 14:11:22.373667  168669 request.go:621] Throttling request took 1.13239322s, request: GET:https://api.crc.testing:6443/apis/rhjmc.redhat.com/v1beta1?timeout=32s
basic:
	Spec Block Exists                   : pass
	CR: containerjfr
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"checkspectest"

	Status Block Exists                 : pass
	CR: containerjfr
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"checkstatustest"
	Suggestions: 
		Add a 'status' field to your Custom Resource

	Writing into CRs has an effect      : pass
	CR: containerjfr
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"writingintocrshaseffecttest"

	Spec Block Exists                   : pass
	CR: example-flightrecorder
	Labels: 
		"test":"checkspectest"
		"necessity":"required"
		"suite":"basic"

	Status Block Exists                 : pass
	CR: example-flightrecorder
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"checkstatustest"
	Suggestions: 
		Add a 'status' field to your Custom Resource

	Writing into CRs has an effect      : pass
	CR: example-flightrecorder
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"writingintocrshaseffecttest"

	Spec Block Exists                   : pass
	CR: example-recording
	Labels: 
		"test":"checkspectest"
		"necessity":"required"
		"suite":"basic"

	Status Block Exists                 : pass
	CR: example-recording
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"checkstatustest"
	Suggestions: 
		Add a 'status' field to your Custom Resource

	Writing into CRs has an effect      : pass
	CR: example-recording
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"writingintocrshaseffecttest"

olm:
	Bundle Validation Test              : pass
	CR: containerjfr
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"bundlevalidationtest"

	Provided APIs have validation       : pass
	CR: containerjfr
	Labels: 
		"suite":"olm"
		"test":"crdshavevalidationtest"
		"necessity":"required"

	Owned CRDs have resources listed    : fail
	CR: containerjfr
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"crdshaveresourcestest"
	Suggestions: 
		If it would be helpful to an end-user to understand or troubleshoot your CR, consider adding resources [secrets/v1 configmaps/v1 containerjfrs/v1alpha1 deployments/v1 services/v1 servicemonitors/v1 persistentvolumeclaims/v1 routes/v1 pods/v1 flightrecorders/v1alpha2 recordings/v1alpha2 replicasets/v1] to the resources section for owned CRD ContainerJFR

	Spec fields with descriptors        : fail
	CR: containerjfr
	Labels: 
		"test":"specdescriptorstest"
		"necessity":"required"
		"suite":"olm"
	Suggestions: 
		Add a spec descriptor for minimal

	Status fields with descriptors      : fail
	CR: containerjfr
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"statusdescriptorstest"

	Bundle Validation Test              : pass
	CR: example-flightrecorder
	Labels: 
		"test":"bundlevalidationtest"
		"necessity":"required"
		"suite":"olm"

	Provided APIs have validation       : pass
	CR: example-flightrecorder
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"crdshavevalidationtest"

	Owned CRDs have resources listed    : fail
	CR: example-flightrecorder
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"crdshaveresourcestest"
	Suggestions: 
		If it would be helpful to an end-user to understand or troubleshoot your CR, consider adding resources [pods/v1 configmaps/v1 recordings/v1alpha2 flightrecorders/v1alpha2 servicemonitors/v1 containerjfrs/v1alpha1 replicasets/v1 deployments/v1 services/v1] to the resources section for owned CRD FlightRecorder

	Spec fields with descriptors        : fail
	CR: example-flightrecorder
	Labels: 
		"suite":"olm"
		"test":"specdescriptorstest"
		"necessity":"required"
	Suggestions: 
		Add a spec descriptor for recordingSelector

	Status fields with descriptors      : fail
	CR: example-flightrecorder
	Labels: 
		"suite":"olm"
		"test":"statusdescriptorstest"
		"necessity":"required"

	Bundle Validation Test              : pass
	CR: example-recording
	Labels: 
		"test":"bundlevalidationtest"
		"necessity":"required"
		"suite":"olm"

	Provided APIs have validation       : pass
	CR: example-recording
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"crdshavevalidationtest"

	Owned CRDs have resources listed    : fail
	CR: example-recording
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"crdshaveresourcestest"
	Suggestions: 
		If it would be helpful to an end-user to understand or troubleshoot your CR, consider adding resources [pods/v1 flightrecorders/v1alpha2 containerjfrs/v1alpha1 replicasets/v1 deployments/v1 configmaps/v1 recordings/v1alpha2 services/v1 servicemonitors/v1] to the resources section for owned CRD Recording

	Spec fields with descriptors        : fail
	CR: example-recording
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"specdescriptorstest"
	Suggestions: 
		Add a spec descriptor for archive
		Add a spec descriptor for duration
		Add a spec descriptor for eventOptions
		Add a spec descriptor for flightRecorder
		Add a spec descriptor for name

	Status fields with descriptors      : fail
	CR: example-recording
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"statusdescriptorstest"


make: *** [Makefile:113: scorecard] Error 1
```

`gingko` succeeded and `scorecard` failed, but the test failures here look the same or similar to what we currently have in upstream `main` if I'm not mistaken.

**Comment by ebaron** [comment]
That scorecard result seems fine for now, there are similar failures in main, as you said. I filed #150 to fix the issues there.

File: go.mod [hunk 1]
```diff
@@ -2,53 +2,22 @@ module github.com/rh-jmc-team/container-jfr-operator
 
 require (
 	github.com/go-logr/logr v0.2.1-0.20200730175230-ee2de8da5be6
-	github.com/jetstack/cert-manager v1.0.2
+	github.com/jetstack/cert-manager v1.1.0
```
Comment (ebaron) [inline go.mod:5]: We should update the Makefile's `CERT_MANAGER_VERSION` too

Comment (andrewazores) [reply go.mod:5]: Ah yes, fix committed

**Review by ebaron — DISMISSED** [review]
Looks good to me!

**Comment by andrewazores** [comment]
Rebased - the conflicts seemed pretty minor and easy to resolve. `make test` still finishes with the same results as before.

File: cmd/manager/main.go [hunk 2]
```diff
@@ -128,8 +129,10 @@ func main() {
 
 	// Create a new Cmd to provide shared dependencies and start components
 	mgr, err := manager.New(cfg, manager.Options{
-		Namespace:          namespace,
-		MapperProvider:     restmapper.NewDynamicRESTMapper,
+		Namespace: namespace,
+		MapperProvider: func(c *rest.Config) (meta.RESTMapper, error) {
+			return apiutil.NewDynamicRESTMapper(c)
+		},
```
Comment (ebaron) [inline cmd/manager/main.go:135]: One question I forgot to ask last time. Does this work?
```suggestion
		MapperProvider: apiutil.NewDynamicRESTMapper,
```

Comment (andrewazores) [reply cmd/manager/main.go:135]: I would think so, but:

`cmd/manager/main.go:133:3: cannot use apiutil.NewDynamicRESTMapper (type func(*rest.Config, ...apiutil.DynamicRESTMapperOption) (meta.RESTMapper, error)) as type func(*rest.Config) (meta.RESTMapper, error) in field value`

Comment (ebaron) [reply cmd/manager/main.go:135]: Ah makes sense. Thanks

**Review by ebaron — APPROVED** [review]

**Pull request merged**